readthedocs/builds/utils.py

```diff
@@ -45,7 +45,7 @@ def get_bitbucket_username_repo(version, repo_url=None):
 
 def get_vcs_version_slug(version):
     slug = None
-    if version.slug == LATEST:
+    if version.slug in NON_REPOSITORY_VERSIONS:
```

Comment:
Not sure if this logic will work. The logic that is after here only applies to 'latest', and will just end up with stable linking to the save version as latest.

Stable is a different branch/tag than latest, and should be treated differently.